### PR TITLE
[wasm][debugger] Removing console.debug message helper to make debugger work.

### LINKED ISSUE #63703: [wasm] Do not print mono_wasm_runtime_ready to console
Do not print `mono_wasm_runtime_ready fe00e07a-5519-4dfe-b35a-f867dbaf2e28` to console. 
It's looks quite bad on NodeJS.

### PR DESCRIPTION
**Pull request by thaystg**
For debugging on chrome it was already unused, if we just remove the message and don't change anything else the debugging continues working, using chrome as ide, using visual studio and also visual studio code.
On firefox it was using the message, but I changed to have the same behavior that has for chrome.
For the debugger tests I created a new message named DotnetDebugger.runTests which will be called when the page is ready to start running the tests.

also:
- renaming wrong file name for firefox
- fix pause in a breakpoint on firefox

Fixes https://github.com/dotnet/runtime/issues/63703

### CODE REVIEW

File: src/mono/wasm/debugger/tests/debugger-test/debugger-driver.html [hunk 1]
```diff
@@ -17,6 +17,7 @@
 				this.run_all = getDotnetRuntime(0).INTERNAL.mono_bind_static_method ("[debugger-test] DebuggerTest:run_all");
 				this.static_method_table = {};
 				console.log ("ready");
+				console.debug("mono_wasm_runtime_ready", "fe00e07a-5519-4dfe-b35a-f867dbaf2e28");
```
Comment (pavelsavara) [inline src/mono/wasm/debugger/tests/debugger-test/debugger-driver.html:20]: Is this only necessary in unit tests ?

Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test/debugger-driver.html:20]: yes

File: src/mono/wasm/runtime/debug.ts [hunk 2]
```diff
@@ -45,8 +45,6 @@ export function mono_wasm_runtime_ready(): void {
         // eslint-disable-next-line no-debugger
         debugger;
     else
```
Comment (lewing) [inline src/mono/wasm/runtime/debug.ts:47]:
```suggestion
```

File: src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs [hunk 3]
```diff
@@ -318,6 +319,16 @@ protected override async Task<bool> AcceptEvent(SessionId sessionId, JObject arg
             case "target-available-form":
                 {
                     OnDefaultContextUpdate(sessionId, new FirefoxExecutionContext(new MonoSDBHelper (this, logger, sessionId), 0, args["target"]["consoleActor"].Value<string>()));
+                    var ctx = GetContextFixefox(sessionId);
+                    ctx.GlobalName = args["target"]["actor"].Value<string>();
+                    ctx.ThreadName = args["target"]["threadActor"].Value<string>();
+                    ResetCmdId();
+                    if (await IsRuntimeAlreadyReadyAlready(sessionId, token))
```
Comment (pavelsavara) [inline src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs:326]: funny name `IsRuntimeAlreadyReadyAlready` :-D

**Comment by thaystg** [comment]
Fixes https://github.com/dotnet/runtime/issues/63703

**Comment by radical** [comment]
just fyi, `runtime-wasm` isn't needed for debugger tests specifically. This is because the debugger tests job is triggered by any changes in `src/mono/wasm/debugger`. But if you have changes anywhere else for which you want to run debugger-tests, then `runtime-wasm` could be useful.

File: src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html [hunk 4]
```diff
@@ -6,6 +6,7 @@
 	<script type='text/javascript'>
 		var App = {
 			init: function () {
+				console.debug ("mono_wasm_runtime_ready", "fe00e07a-5519-4dfe-b35a-f867dbaf2e28");
```
Comment (lewing) [inline src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html:9]: Is something depending on this behavior? if so we should fix it?

File: src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs (hunk 3 above)
Comment (lewing) [reply src/mono/wasm/debugger/BrowserDebugProxy/Firefox/FirefoxMonoProxy.cs:326]: I uh... https://www.youtube.com/watch?v=f8F_smVEuLg

File: src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html (hunk 4 above)
Comment (thaystg) [reply src/mono/wasm/debugger/tests/debugger-test/wasm-page-without-assets.html:9]: The firefox debugger tests.
But @radical changed other things and now this PR is old, I need to merge main and also I'm planning to fix this:
https://github.com/dotnet/runtime/issues/73651 in the same PR, then we can totally remove this message probably.

**Comment by pavelsavara** [comment]
Could you also please delete https://github.com/dotnet/runtime/blob/83ce1253cf71348776fec4464e3fad1277fac7f0/src/mono/sample/wasm/browser-bench/README.md?plain=1#L28

**Comment by pavelsavara** [comment]
If we are no longer sending that message, could we also delete 

https://github.com/dotnet/runtime/blob/83ce1253cf71348776fec4464e3fad1277fac7f0/src/mono/wasm/debugger/BrowserDebugProxy/DevToolsHelper.cs#L334-L335

And 

https://github.com/dotnet/runtime/blob/83ce1253cf71348776fec4464e3fad1277fac7f0/src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs#L107-L127

I'm bit confused about if it's possible.

**Comment by thaystg** [comment]
> If we are no longer sending that message, could we also delete
> 
> https://github.com/dotnet/runtime/blob/83ce1253cf71348776fec4464e3fad1277fac7f0/src/mono/wasm/debugger/BrowserDebugProxy/DevToolsHelper.cs#L334-L335
> 
> And
> 
> https://github.com/dotnet/runtime/blob/83ce1253cf71348776fec4464e3fad1277fac7f0/src/mono/wasm/debugger/BrowserDebugProxy/MonoProxy.cs#L107-L127
> 
> I'm bit confused about if it's possible.

After your comment I tried another approach which is completely removing the message. :)

**Review by pavelsavara — APPROVED** [review]
I guess you tested it manually too?

**Comment by thaystg** [comment]
> I guess you tested it manually too?

Yes, I did :)

**Comment by thaystg** [comment]
@ilonatommy can you double check if I didn't break anything, please? Test it on blazor using VS, using VSCode and using ctrl-shift-d from chrome?

**Comment by ilonatommy** [comment]
> @ilonatommy can you double check if I didn't break anything, please? Test it on blazor using VS, using VSCode and using ctrl-shift-d from chrome?

Working. Checked:
- [X] VS debugger
- [X] VS Code
- [X] Chrome Dev tools

**Review by ilonatommy — APPROVED** [review]
Looks cool!

**Pull request merged**